src/passes/LegalizeJSInterface.cpp

```diff
@@ -165,14 +132,33 @@ struct LegalizeJSInterface : public Pass {
             // call
             return;
           }
+
           replaceCurrent(
             Builder(*getModule())
               .makeCall(
                 iter->second, curr->operands, curr->type, curr->isReturn));
         }
+
+        void visitRefFunc(RefFunc* curr) {
+          auto iter = illegalImportsToLegal->find(curr->func);
+          if (iter == illegalImportsToLegal->end()) {
+            return;
+          }
+
+          auto newName = iter->second;
+          if (newName == getFunction()->name) {
+            // inside the stub function itself, is the one safe place to do the
+            // call
```

Comment:
Why? I guess this function will be deleted anyway so doesn't matter, but is there a reason to make this an exception?

Reply:
Good catch! I think this code is pointless... No one will call the illegal function after this transform anyhow. I copied it from above as Call has the same problem to solve, but it's not needed in both places... I'll remove it.